internal/node/controller/auth.go

```diff
@@ -103,3 +109,186 @@ func getUserCertAndInfo(controller NodeController, username string) (*x509.Certi
+
+type BFFChallengeRequest struct {
+	DID string `json:"did"`
+}
+
+type FriendStore map[string]*Friend
+
+type Friend struct {
+	DID       string `json:"did"`
+	PublicKey crypto.PublicKey
+}
+
+type FriendRequest struct {
+	DID                string `json:"did"`
+	PublicKeyMultibase string `json:"public_key_multibase"`
+}
+
+type BFFProvider struct {
+	challengePersister bffauth.ChallengeSessionPersister
+	friends            FriendStore
+	signingKey         []byte
+}
+
+func NewBFFProvider(challengePersister bffauth.ChallengeSessionPersister, friends FriendStore, signingKey []byte) *BFFProvider {
+	return &BFFProvider{
+		challengePersister: challengePersister,
+		friends:            friends,
+		signingKey:         signingKey,
+	}
+}
+
+func (p *BFFProvider) GetRoutes() []api.Route {
+	return []api.Route{
+		newRoute(http.MethodPost, "/node/bff/challenge", p.handleChallenge),
+		newRoute(http.MethodPost, "/node/bff/auth", p.handleAuth),
+		newRoute(http.MethodPost, "/node/bff/add_friend", p.handleAddFriend),
+		newRoute(http.MethodGet, "/node/bff/test", p.handleTest),
+	}
+}
+
+func (p *BFFProvider) handleChallenge(w http.ResponseWriter, r *http.Request) {
+	var req BFFChallengeRequest
+	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
+		http.Error(w, "Invalid request body", http.StatusBadRequest)
+		return
+	}
+
+	_, ok := p.friends[req.DID]
+	if !ok {
+		http.Error(w, fmt.Sprintf("Friend with DID %s not found", req.DID), http.StatusNotFound)
+		return
+	}
+
+	challenge, err := bffauth.GenerateChallenge()
+	if err != nil {
+		http.Error(w, "Failed to generate challenge", http.StatusInternalServerError)
+		return
+	}
+
+	// Create session with UUID
+	sessionID := uuid.New().String()
+	session := &bffauth.ChallengeSession{
+		SessionID: sessionID,
+		DID:       req.DID,
+		Challenge: challenge,
+		ExpiresAt: time.Now().Add(15 * time.Minute),
+	}
+	err = p.challengePersister.SaveSession(session)
+	if err != nil {
+		http.Error(w, "Failed to save session", http.StatusInternalServerError)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	err = json.NewEncoder(w).Encode(map[string]string{
+		"challenge": challenge,
+		"session":   sessionID,
+	})
+	if err != nil {
+		http.Error(w, "Failed to encode response", http.StatusInternalServerError)
+		return
+	}
+}
+
+func (p *BFFProvider) handleAuth(w http.ResponseWriter, r *http.Request) {
+	var req struct {
+		SessionID string `json:"session_id"`
+		Proof     string `json:"proof"`
+	}
+	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
+		http.Error(w, "Invalid request body", http.StatusBadRequest)
+		return
+	}
+
+	session, err := p.challengePersister.GetSession(req.SessionID)
+	if err != nil {
+		http.Error(w, "Invalid session", http.StatusBadRequest)
+		return
+	}
+
+	err = p.challengePersister.DeleteSession(req.SessionID)
+	if err != nil {
+		http.Error(w, "Failed to delete session", http.StatusInternalServerError)
+		return
+	}
```

Comment:
should we have a TODO to not delete it immediately? like we probably want to delete it when it's verified OR have an expiry on it right? (otherwise someone else could come in and end your challenge session)